src/plugins/intel_gna/src/transformations/gather_sinking_unary.cpp

```diff
@@ -0,0 +1,205 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/gather_sinking_unary.hpp"
+
+#include <transformations/utils/utils.hpp>
+#include <utility>
+#include <openvino/cc/ngraph/itt.hpp>
+
+#include "openvino/opsets/opset9.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "transformations/gather_sinking_utils.hpp"
+#include "transformations/rt_info/gather_sinking_attr.hpp"
+
+using namespace ov;
+using namespace ov::opset9;
+using namespace ov::pass::pattern;
+using namespace ov::op::util;
+using namespace gather_sinking;
+
+namespace {
+
+using NodePtr = std::shared_ptr<ov::Node>;
+using NodePair = std::pair<NodePtr, NodePtr>;
+
+/**
+ * @brief SwapNodes allows to perform swapping nodes even if there are more than one consumers but has less performance
+ *
+ * @param first_node first node pointer
+ * @param second_node first node pointer
+ * @return NodePair pair of nodes in new order that allows to register them in MatcherPass
+ */
+NodePair SwapNodes(NodePtr first_node, NodePtr second_node) {
+    auto second_node_inputs = second_node->input_values();
+    second_node_inputs[0] = first_node->input_value(0);
+
+    auto new_first_node = second_node->clone_with_new_inputs(second_node_inputs);
+
+    auto first_node_inputs = first_node->input_values();
+    first_node_inputs[0] = new_first_node;
+    auto new_second_node = first_node->clone_with_new_inputs(first_node_inputs);
+
+    new_second_node->set_friendly_name(second_node->get_friendly_name());
+    ov::copy_runtime_info({first_node, second_node}, {new_first_node, new_second_node});
+
+    ov::replace_node(second_node, new_second_node);
+
+    return std::make_pair(new_first_node, new_second_node);
+}
+
+/**
+ * @brief SwapOutputs has much better performance than SwapNodes and covers the most of the real situations
+ *        but cannot work when the consumers count greater than one
+ * @param first_node first node pointer
+ * @param second_node second node pointer
+ * @return NodePair pair of nodes in new order that allows to register them in MatcherPass
+ */
+NodePair SwapOutputs(NodePtr first_node, NodePtr second_node) {
+    const auto first_node_output_names = first_node->output(0).get_names();
```

Comment:
get_default_output()? but for some layers it cannot be used (Split)

Reply:
first_node->output(0) is the same as first_node->get_default_output() but shorter
It is a matter of taste